### PR TITLE
PC-1256: Allow hiding columns

### PR DESCRIPTION
**Pull request by tomecko**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,9 +50,9 @@ A React component for displaying table data (with lot of built-in features)
       <Filter active="?Boolean" disabled="?Boolean" filterKey="?String" label="!String" predicate="!Function" />
     </FiltersGroup>
     …
-    <DataPaginationNav className="?" countThreshold="?Number" onAsyncPageChange="?Function" style="?" />
+    <DataPaginationNav className="?" onAsyncPageChange="?Function" style="?" />
     …
-    <Search className="?" expandable="?Boolean" style="?" />
+    <Search className="?" debounceTime="?Number" expandable="?Boolean" instructions="?ReactNode" onAsyncSearch="?Function" onHideInstructions="?Function" onShowInstructions="?Function" parserNames="?ParserName[]" placeholder="!String" searchParser="?SearchParser" showInstructions="?Boolean" style="?" />
```
Comment (ulfryk) [inline README.md]: :+1:

File: src/components/data-pagination/data-pagination-dropdown.tsx [hunk 2]
```diff
@@ -21,6 +21,8 @@ export interface IDataPaginationDropdownProps<T> {
 const PROP_TYPES: ValidationMap<IDataPaginationDropdownProps<any>> = {
   availableLimits: arrayOf(number).isRequired,
   beforeLimitChange: func,
+  context: instanceOf(DataPanelController).isRequired,
```
Comment (ulfryk) [inline src/components/data-pagination/data-pagination-dropdown.tsx]: this should be passed through context (if possible), see `src/config/context.ts`

see:
- `src/components/highlight.tsx`
- `src/components/data-table/row.tsx`
- …etc

File: README.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ A React component for displaying table data (with lot of built-in features)
   items="!ItemType[]"
   expandedItems="?Set<Id>"
   selectedItems="?Set<Id>"
+  hiddenColumns="?Set<string>"
```
Comment (ulfryk) [inline README.md]: where's `onHiddenColumnsChange` ?

File: src/components/manage-columns/manage-columns.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+import * as React from 'react';
+import { Button, Checkbox } from 'react-bootstrap';
+
+import { DataPanelController } from '../../utility';
+import * as css from './manage-columns.class-names';
+
+import './manage-columns.sass';
+
+interface IManageColumnsProps<T> {
+  readonly context: DataPanelController<T>;
```
Comment (ulfryk) [inline src/components/manage-columns/manage-columns.tsx]: should come from React `context`

File: src/components/manage-columns/manage-columns.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+          bsSize="xs"
+          className={css.ACTION}
+          onClick={() => {
+            this.togglePopover();
+          }}
+        >
+          Manage Columns
+        </Button>
+        {this.state.showPopover ? (
+          <div className={css.POPOVER}>
```
Comment (ulfryk) [inline src/components/manage-columns/manage-columns.tsx]: form ? onSubmit ?

File: README.md (hunk 3 above)
Comment (tomecko) [reply README.md]: now it belongs to `DataTable`

**Pull request merged**